session_2/cricviz/app/models/cricketer.rb

```diff
@@ -68,14 +146,56 @@ def self.import_classical_batters
   # `ActiveRecord::RecordNotFound` exception with the player's name as
   # the message.
   def self.update_innings(batting_scorecard, bowling_scorecard)
-    raise NotImplementedError
+    batting_scorecard.each do |data|
+      player = Cricketer.find_by(name: data[0])
+      if !player
+        raise ActiveRecord::RecordNotFound, "#{data[0]}"
+      else
```

Comment:
We don't need the else as you can think of an exception to be same as break out of the loop.

Something like:
```ruby
player = Cricketer.find_by(name: data[0])
if !player
  raise ActiveRecord::RecordNotFound, "#{data[0]}"
end

if !data[1]
  player.not_out += 1
end

...

player.save
```